Generated zod schemas fail to compile when setting output to a custom dir

This happens when the ZModel triggers generation of a logical prisma client.

```prisma
model User {
    id Int @id
    posts Post[]
}

model Post {
    id Int @id
    author User @relation(fields: [authorId], references: [id])
    authorId Int @default(auth().id)
}
```

```bash
npx zenstack generate --schema=./src/schema.zmodel --output=./src/lib/generated --no-compile
```

```
Error compiling generated code:
lib/generated/zod/input/AmortizationScheduleInput.schema.ts:4:29 - error TS2307: Cannot find module '.zenstack/models' or its corresponding type declarations.

4 import type { Prisma } from '.zenstack/models';
                              ~~~~~~
lib/generated/zod/input/AmortizationScheduleItemInput.schema.ts:4:29 - error TS2307: Cannot find module '.zenstack/models' or its corresponding type declarations.
```

--- a/packages/schema/src/plugins/enhancer/enhance/index.ts
+++ b/packages/schema/src/plugins/enhancer/enhance/index.ts
@@ -95,7 +95,8 @@ export class EnhancerGenerator {
 
         const enhanceTs = this.project.createSourceFile(
             path.join(this.outDir, 'enhance.ts'),
-            `import { type EnhancementContext, type EnhancementOptions, type ZodSchemas, type AuthUser } from '@zenstackhq/runtime';
+            `/* eslint-disable */
+import { type EnhancementContext, type EnhancementOptions, type ZodSchemas, type AuthUser } from '@zenstackhq/runtime';
 import { createEnhancement } from '@zenstackhq/runtime/enhancements';
 import modelMeta from './model-meta';
 import policy from './policy';

--- a/packages/schema/src/plugins/enhancer/index.ts
+++ b/packages/schema/src/plugins/enhancer/index.ts
@@ -24,9 +24,11 @@ const run: PluginFunction = async (model, options, _dmmf, globalOptions) => {
     let prismaClientPath: string | undefined;
     if (dmmf) {
         // a logical client is generated
-        if (typeof options.output === 'string') {
+        if (options.output || globalOptions?.output) {
+            // handle custom output path
+
             // get the absolute path of the prisma client types
-            const prismaClientPathAbs = path.resolve(options.output, 'models');
+            const prismaClientPathAbs = path.resolve(outDir, 'models');
 
             // resolve it relative to the schema path
             prismaClientPath = path.relative(path.dirname(options.schemaPath), prismaClientPathAbs);

--- a/tests/regression/tests/issue-1610.test.ts
+++ b/tests/regression/tests/issue-1610.test.ts
@@ -1,0 +1,56 @@
+import { loadSchema } from '@zenstackhq/testtools';
+describe('issue 1610', () => {
+    it('regular prisma client', async () => {
+        await loadSchema(
+            `
+            model User {
+                id Int @id
+                posts Post[]
+            }
+            
+            model Post {
+                id Int @id
+                author User @relation(fields: [authorId], references: [id])
+                authorId Int
+            }
+            `,
+            { fullZod: true, output: './lib/zen' }
+        );
+    });
+
+    it('logical prisma client', async () => {
+        await loadSchema(
+            `
+            model User {
+                id Int @id
+                posts Post[]
+            }
+            
+            model Post {
+                id Int @id
+                author User @relation(fields: [authorId], references: [id])
+                authorId Int @default(auth().id)
+            }
+            `,
+            { fullZod: true, output: './lib/zen' }
+        );
+    });
+
+    it('no custom output', async () => {
+        await loadSchema(
+            `
+            model User {
+                id Int @id
+                posts Post[]
+            }
+            
+            model Post {
+                id Int @id
+                author User @relation(fields: [authorId], references: [id])
+                authorId Int @default(auth().id)
+            }
+            `,
+            { fullZod: true, preserveTsFiles: true }
+        );
+    });
+});

